### PR TITLE
virtualbox: 6.1.34 -> 6.1.36

### PR DESCRIPTION
**Pull request by andresilva**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Yarny0** [comment]
#### Review per https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-package-updates

##### Reviewed points

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86-64
- [x] executables tested on x86-64
- [ ] all depending packages build

More

- All vm tests (including the unfree extension pack test) pass with current default kernel 5.15.  To be sure, I also tested with 5.18 (see comment below regarding guest additions) as well as with older kernels:

  Test                    |  4.9  |  4.14  |  4.19  | 5.4 | 5.10 | 5.15 | 5.18
  ------------------------|:-----:|:------:|:------:|:---:|:----:|:----:|:----:
  `net-hostonlyif`        | FAIL  |  FAIL  |  FAIL  | OK  |  OK  |  OK  |  OK
  `systemd-detect-virt`   | FAIL  |  FAIL  |  FAIL  | OK  |  OK  |  OK  |  OK
  `host-usb-permissions`  |  OK   |   OK   |   OK   | OK  |  OK  |  OK  |  OK
  `headless`              | FAIL  |  FAIL  |  FAIL  | OK  |  OK  |  OK  |  OK
  `simple-cli`            | FAIL  |  FAIL  |  FAIL  | OK  |  OK  |  OK  |  OK
  `simple-gui`            | FAIL  |  FAIL  |  FAIL  | OK  |  OK  |  OK  |  OK
  `enable-extension-pack` | FAIL  |  FAIL  |  FAIL  | OK  |  OK  |  OK  |  OK

  These failures aren't a regression, as the matrix is the same with the old virtualbox 6.1.34 version (up to the 5.18 column).

  Those failures happend inside the guest's initrd, so VirtualBox itself should even work on older kernels.

- Tested virtualbox on real hardware (only with kernel 5.15) to verify that the bug from https://github.com/NixOS/nixpkgs/issues/179210 didn't return.  Luckily, it didn't!

- Note: I prepared the very same commit (up the the proposed changed regarding guest additions below).  The tests documented above have been performed based on an older version of the nixos-unstable branch: 5a0e0d73b944157328d54c4ded1cf2f0146a86a5.

##### Possible improvements

- The commit could fix https://github.com/NixOS/nixpkgs/issues/182985 , but this requires the following additional change:

  ```diff
  diff --git a/pkgs/applications/virtualization/virtualbox/guest-additions/default.nix b/pkgs/applications/virtualization/virtualbox/guest-additions/default.nix
  index 538ebfa78cf..75684cb3b99 100644
  --- a/pkgs/applications/virtualization/virtualbox/guest-additions/default.nix
  +++ b/pkgs/applications/virtualization/virtualbox/guest-additions/default.nix
  @@ -148,6 +148,6 @@ in stdenv.mkDerivation rec {
       license = "GPL";
       maintainers = [ lib.maintainers.sander ];
       platforms = [ "i686-linux" "x86_64-linux" ];
  -    broken = kernel.kernelAtLeast (if stdenv.hostPlatform.is32bit then "5.10" else "5.17");
  +    broken = stdenv.hostPlatform.is32bit && (kernel.kernelAtLeast "5.10");
     };
   }
  ```

  I suggest adding this change to the commit.  Besides that, I consider this commit ready for merging.

##### Comments

Notifying maintainers: @svanderburg , @cdepillabout (extension pack)

**Comment by andresilva** [comment]
@Yarny0 thanks for the very thorough review and comments! I have updated the `broken` check as suggested.

File: pkgs/applications/virtualization/virtualbox/default.nix [hunk 1]
```diff
@@ -98,15 +98,6 @@ in stdenv.mkDerivation {
     ./qtx11extras.patch
     # https://github.com/NixOS/nixpkgs/issues/123851
     ./fix-audio-driver-loading.patch
-    # NOTE: both patches below should be removed when updating to 6.1.35
-    # https://www.virtualbox.org/ticket/20914#comment:15
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/virtualbox/default.nix:103]: Please remove the unused fetchpatch from inputs.

**Review by SuperSandro2000 — COMMENTED** [review]
.

**Pull request merged**